statsmodels/gam/tests/test_gam.py

```diff
@@ -0,0 +1,808 @@
+    return loglike - alpha * itg, loglike, itg
+
+
+def test_gam_penalty():
+    """
+    test the func method of the gam penalty
+    :return:
+    """
+    pol, y = polynomial_sample_data()
+    univ_pol = pol.smoothers[0]
+    alpha = 1
+    gp = UnivariateGamPenalty(alpha=alpha, univariate_smoother=univ_pol)
+
+    for _ in range(10):
+        params = np.random.randint(-2, 2, 4)
+        gp_score = gp.func(params)
+        itg = integral(params)
+        assert_allclose(gp_score, itg, atol=1.e-1)
+
+
+def test_gam_gradient():
+    # test the gam gradient for the example polynomial
+    np.random.seed(1)
+    pol, y = polynomial_sample_data()
+
+    alpha = 1
+    smoother = pol.smoothers[0]
+    gp = UnivariateGamPenalty(alpha=alpha, univariate_smoother=smoother)
+
+    for _ in range(10):
+        params = np.random.uniform(-2, 2, 4)
+        params = np.array([1, 1, 1, 1])
+        gam_grad = gp.deriv(params)
+        grd = grad(params)
+
+        assert_allclose(gam_grad, grd, rtol=1.e-2, atol=1.e-2)
+
+
+def test_gam_hessian():
+    # test the deriv2 method of the gam penalty
+    np.random.seed(1)
+    pol, y = polynomial_sample_data()
+    univ_pol = pol.smoothers[0]
+    alpha = 1
+    gp = UnivariateGamPenalty(alpha=alpha, univariate_smoother=univ_pol)
+
+    for _ in range(10):
+        params = np.random.randint(-2, 2, 5)
+        gam_der2 = gp.deriv2(params)
+        hess = hessian(params)
+        hess = np.flipud(hess)
+        hess = np.fliplr(hess)
+        assert_allclose(gam_der2, hess, atol=1.e-13, rtol=1.e-3)
+
+
+def test_approximation():
+    np.random.seed(1)
+    poly, y = polynomial_sample_data()
+    alpha = 1
+    for _ in range(10):
+        params = np.random.uniform(-1, 1, 4)
+        cost, err, itg = cost_function(params, poly, y, alpha)
+        glm_gam = GLMGam(y, smoother=poly, alpha=alpha)
+        # TODO: why do we need pen_weight=1
+        gam_loglike = glm_gam.loglike(params, scale=1, pen_weight=1)
+        assert_allclose(err - itg, cost, rtol=1e-10)
+        assert_allclose(gam_loglike, cost, rtol=0.1)
+
+
+def test_gam_glm():
+    cur_dir = os.path.dirname(os.path.abspath(__file__))
+    file_path = os.path.join(cur_dir, "results", "prediction_from_mgcv.csv")
+    data_from_r = pd.read_csv(file_path)
+    # dataset used to train the R model
+    x = data_from_r.x.as_matrix()
+    y = data_from_r.y.as_matrix()
+
+    df = [10]
+    degree = [3]
+    bsplines = BSplines(x, degree=degree, df=df, include_intercept=True)
+    # y_mgcv is obtained from R with the following code
+    # g = gam(y~s(x, k = 10, bs = "cr"), data = data, scale = 80)
+    y_mgcv = np.asarray(data_from_r.y_est)
+
+    # alpha = 1000
+    alpha = 0.03
+    alpha = 0.0001
+    alpha = 0.1
+
+    glm_gam = GLMGam(y, smoother=bsplines, alpha=alpha)
+    res_glm_gam = glm_gam.fit(method='bfgs', max_start_irls=0,
+                              disp=1, maxiter=10000, maxfun=5000)
```

Comment:
looks like a duplicate 
maybe I had needed it for start_params with "nm", bfgs works fine here